otel.go

```diff
@@ -55,42 +65,49 @@ func (lk *logkeeper) initOtel(ctx context.Context) error {
 		grip.Error(errors.Wrap(err, "otel error"))
 	}))
 
-	lk.tracer = tp.Tracer(packageName)
-
-	lk.closers = append(lk.closers, closerOp{
+	closers = append(closers, closerOp{
 		name: "tracer provider shutdown",
 		closerFn: func(ctx context.Context) error {
 			catcher := grip.NewBasicCatcher()
 			catcher.Wrap(tp.Shutdown(ctx), "trace provider shutdown")
 			catcher.Wrap(traceExporter.Shutdown(ctx), "trace exporter shutdown")
-			catcher.Wrap(lk.otelGrpcConn.Close(), "closing gRPC connection")
-
 			return catcher.Resolve()
 		},
 	})
+	useCustomName = true
+}
 
-	return nil
+func initTracer(name string) otelTrace.Tracer {
+	var tracerName string
+	if useCustomName {
+		tracerName = fmt.Sprintf(packageName, name)
+	} else {
+		tracerName = defaultName
+	}
+	return otel.GetTracerProvider().Tracer(tracerName)
 }
 
-func (lk *logkeeper) Close(ctx context.Context) {
-	catcher := grip.NewBasicCatcher()
-	for idx, closer := range lk.closers {
-		if closer.closerFn == nil {
-			continue
-		}
+func Close(ctx context.Context) {
+	closer.Do(func() {
```

Comment:
are we worried this will be called more than once?
And if it is called more than once would it be a problem?

Reply:
I added it for safety, but since we explicitly call it, I removed it

Reply:
we may see errors if it is called twice. otel may complain that we are trying to close a closed connection